### PR TITLE
CDAP-4121 remove usage of Id.NotificationFeed

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DUT5148

File: cdap-notifications/src/main/java/co/cask/cdap/notifications/service/kafka/KafkaNotificationService.java [hunk 1]
```diff
@@ -218,8 +219,9 @@ public void onReceived(Iterator<FetchedMessage> messages) {
           KafkaMessage decodedMessage = KafkaMessageCodec.decode(payload);
           try {
             LOG.trace("Decoded notification from Kafka: {}", decodedMessage);
-            notificationReceived(KafkaNotificationUtils.getMessageFeed(decodedMessage.getMessageKey()),
-                                 decodedMessage.getNotificationJson());
+            NotificationFeedId feedId =
+              NotificationFeedId.fromIdParts(Splitter.on('.').split(decodedMessage.getMessageKey()));
```
Comment (anwar6953) [inline cdap-notifications/src/main/java/co/cask/cdap/notifications/service/kafka/KafkaNotificationService.java:223]: can you use fromString method here instead? and toString on the other side? Or is that too implicit?

Comment (albertshau) [reply cdap-notifications/src/main/java/co/cask/cdap/notifications/service/kafka/KafkaNotificationService.java:223]: if there are notifications in kafka that have not been consumed, they will be incompatible.

File: cdap-notifications/src/main/java/co/cask/cdap/notifications/service/kafka/KafkaNotificationService.java [hunk 2]
```diff
@@ -236,4 +238,20 @@ public void finished() {
       LOG.info("Subscription to topic partition {} finished.", topicPartition);
     }
   }
+
+
+  /**
+   * Map a {@link NotificationFeedId} to a Kafka topic partition.
+   *
+   * @param feed {@link NotificationFeedId} object
+   * @param topic kafka topic
```
Comment (sreevatsanraman) [inline cdap-notifications/src/main/java/co/cask/cdap/notifications/service/kafka/KafkaNotificationService.java:246]: nit: order of parameter is different. can you fix?

**Comment by sreevatsanraman** [comment]
one minor comment. Fix and merge.

**Pull request merged**